eth-custodian/contracts/EthCustodianProxy.sol

```diff
@@ -97,7 +98,7 @@ contract EthCustodianProxy is
         bytes calldata proofData,
         uint64 proofBlockHeight
     ) external {
-        if (proofBlockHeight > migrationBlockHeight) {
+        if (BlockHeightFromProofExtractor.getBlockHeightFromProof(proofData) > migrationBlockHeight) {
```

Comment:
I propose to add an argument for optimisation purposes, so the height could be verified only if the front pass `isPreMigration`, it should be safe because it is still possible to finalise the transfers directly with the implementation contract.

Reply:
fixed: https://github.com/Near-One/eth-connector/pull/67/commits/4098d957643641526f97416b04a59af43ce343bd

Reply:
I thought a bit more about it, and I think it will be more convenient to pass `receiptBlockHeight`. In this way, the front won't care about this check, and just pass the receipt height. The contract itself will do the check if it is a pre migration tx height

Reply:
I have a thought that you can always pass 'true' here without thinking, and then it will definitely work. And if there's already a `receiptBlockHeight` parameter, it shouldn't be difficult to check it on the front end. Pierre already added MigrationHeight https://github.com/aurora-is-near/rainbow-bridge-client/blob/0f7f5929d99b0323d8301b70833042136470fde1/packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts#L92

Reply:
@olga24912 yeh, I know that both of the solutions should work, but from my point of view, passing receipt height looks better and more robust.
Anyway, it is subjective thing, so let's ask @kiseln and @paouvrard, what do they think.

Reply:
No big difference but I think I prefer receiptBlockHeight: it is a little simpler for the caller of that function because you can call it without being aware of the migration block height as long as you know which account generated the receipt.